website/docs/r/application_gateway.html.markdown

```diff
@@ -0,0 +1,344 @@
+
+
+The `request_routing_rule` block supports:
+
+* `name` - (Required) User defined name for a request routing rule.
+
+* `rule_type' - (Required) Routing rule type. Valid values are:
+  * `Basic`
+  * `PathBasedRouting`
+
+* `http_listener_name` - (Required) Reference to `http_listener`.
+
+* `backend_address_pool_name` - (Optional) Reference to `backend_address_pool_name`. Valid for Basic Rule only.
+
+* `backend_http_settings_name` - (Optional) Reference to `backend_http_settings`. Valid for Basic Rule only.
+
+* `url_path_map_name` - (Optional) Reference to `url_path_map`. Valid for PathBasedRouting Rule only.
+
+The `url_path_map` block supports:
+
+* `name` - (Required) User defined name for a url path map.
+
+* `default_backend_address_pool_name` - (Required) Reference to `backend_address_pool_name`.
+
+* `default_backend_http_settings_name` - (Required) Reference to `backend_http_settings`.
+
+* `path_rule` - (Required) List of pathRules. pathRules are order sensitive. Are applied in order they are specified.
+
+The `path_rule` block supports:
+
+* `name` - (Required) User defined name for a path rule.
+
+* `paths` - (Required) The list of path patterns to match. Each must start with / and the only place a * is allowed is at the end following a /. The string fed to the path matcher does not include any text after the first ? or #, and those chars are not allowed here.
+
+* `backend_address_pool_name` - (Required) Reference to `backend_address_pool_name`. 
+
+* `backend_http_settings_name` - (Required) Reference to `backend_http_settings`.
+
+The `authentication_certificate` block supports:
+
+* `name` - (Required) User defined name for an authentication certificate.
+
+* `data` - (Required) Base-64 encoded cer certificate. Only applicable in PUT Request.
+
+The `ssl_certificate` block supports:
+
+* `name` - (Required) User defined name for an SSL certificate.
+
+* `data` - (Required) Base-64 encoded Public cert data corresponding to pfx specified in data. Only applicable in GET request.
+
+* `password` - (Required) Password for the pfx file specified in data. Only applicable in PUT request.
+
+The `waf_configuration` block supports:
+
+* `name` - (Required) User defined name for a web application firewall.
+
+* `firewall_mode` - (Required) Firewall mode.  Valid values are:
+  * `Detection`
+  * `Prevention`
+  
+* `rule_set_type` - (Required) Rule set type.  Must be set to `OWASP`
+
+* `rule_set_version` - (Required) Ruleset version. Supported values: 
+  * `2.2.9`
+  * `3.0`
+
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The application gatewayConfiguration ID.
+
+* `name` - The name of the application gateway.
+
+* `resource_group_name` - The name of the resource group in which to create the application gateway.
+
+* `location` - The location/region where the application gateway is created
+
+
+
+## Import
```

Comment:
It doesn't look like we are importing this resource. Do you mind adding 
```
Importer: &schema.ResourceImporter{
  State: schema.ImportStatePassthrough,
},
``` 
and writing an import test?